### ISSUE TITLE
DomainName is not a valid property for CodeArtifact's Repositories

### ISSUE THREAD

**Issue by sledigabel** [aws-cloudformation/cfn-python-lint#1776]
*cfn-lint version: 0.40.0*

*cfn-lint reports that DomainName is not a valid property for CodeArtifact's Repositories*

The following snippet is taken from the [CodeArtifact CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codeartifact-repository.html):
```yaml
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  MyCodeArtifactDomain:
    Type: 'AWS::CodeArtifact::Domain'
    Properties:
      DomainName: "my-domain"
  MyCodeArtifactUpstreamRepository:
    Type: 'AWS::CodeArtifact::Repository'
    Properties:
      RepositoryName: "my-upstream-repo"
      DomainName: !GetAtt MyCodeArtifactDomain.Name
      ExternalConnections:
        - public:npmjs
  MyCodeArtifactRepository:
    Type: 'AWS::CodeArtifact::Repository'
    Properties:
      RepositoryName: "my-repo"
      DomainName: !GetAtt MyCodeArtifactDomain.Name
      Upstreams:
        - !GetAtt MyCodeArtifactUpstreamRepository.Name
```

This yaml is valid and processed by CloudFormation successfully, but is tagged as invalid by `cfn-lint`:
```
$ cfn-lint -u # ensure we use the latest version of the specs
$ cfn-lint cloudformation_codeartifact_cfn_lint.yml
E3002 Invalid Property Resources/MyCodeArtifactUpstreamRepository/Properties/DomainName
cloudformation_codeartifact_cfn_lint.yml:12:7

E3002 Invalid Property Resources/MyCodeArtifactRepository/Properties/DomainName
cloudformation_codeartifact_cfn_lint.yml:19:7
```

The rest of the yaml document seems to be parsed correctly.
Looking closer, it appears that the `DomainName` property is also missing from the [documentation in Cloudformation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codeartifact-repository.html).
![image](https://user-images.githubusercontent.com/1921711/98818035-e9818680-2422-11eb-96b2-d99f8c46435b.png)

**Comment by PatMyron** [comment]
https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeartifact/pull/38#discussion_r518897362

**Comment by PatMyron** [comment]
[just merged in yesterday](https://github.com/aws-cloudformation/cfn-python-lint/pull/1831), will be fixed in the next release
